confignode/src/main/java/org/apache/iotdb/confignode/consensus/response/DataPartitionDataSet.java

```diff
@@ -20,16 +20,110 @@
 package org.apache.iotdb.confignode.consensus.response;
 
 import org.apache.iotdb.commons.partition.DataPartition;
+import org.apache.iotdb.confignode.rpc.thrift.TDataPartitionResp;
+import org.apache.iotdb.confignode.rpc.thrift.TRegionReplicaSet;
+import org.apache.iotdb.confignode.rpc.thrift.TSeriesPartitionSlot;
+import org.apache.iotdb.confignode.rpc.thrift.TTimePartitionSlot;
 import org.apache.iotdb.consensus.common.DataSet;
+import org.apache.iotdb.service.rpc.thrift.EndPoint;
+import org.apache.iotdb.service.rpc.thrift.TSStatus;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
 public class DataPartitionDataSet implements DataSet {
-  private DataPartition dataPartitionInfo;
 
-  public DataPartition getDataPartitionInfo() {
-    return dataPartitionInfo;
+  private TSStatus status;
+
+  private DataPartition dataPartition;
+
+  public TSStatus getStatus() {
+    return status;
+  }
+
+  public void setStatus(TSStatus status) {
+    this.status = status;
+  }
+
+  public DataPartition getDataPartition() {
+    return dataPartition;
+  }
+
+  public void setDataPartition(DataPartition dataPartition) {
+    this.dataPartition = dataPartition;
   }
 
-  public void setDataPartitionInfos(DataPartition dataPartitionInfo) {
-    this.dataPartitionInfo = dataPartitionInfo;
+  /**
+   * Convert DataPartitionDataSet to TDataPartitionResp
+   *
+   * @param resp TDataPartitionResp
+   */
+  public void convertToRpcDataPartitionResp(TDataPartitionResp resp) {
+    Map<String, Map<TSeriesPartitionSlot, Map<TTimePartitionSlot, List<TRegionReplicaSet>>>>
+        dataPartitionMap = new HashMap<>();
+
+    dataPartition
```

Comment:
The same as me... I'll check the performance of this maps by tools like Arthas later.

Reply:
The data structures used in PartitionTable are in node-commons currently. I have to use them because I need some interfaces defined by myself. So the transition between the thrift structures and the self-defined structures is necessary for present.

Reply:
>  I'll check the performance of this maps by tools like Arthas later.

Looking forward to it.